src/Weeder.hs

```diff
@@ -410,3 +573,33 @@ nameToDeclaration name = do
 
 unNodeAnnotation :: NodeAnnotation -> (String, String)
 unNodeAnnotation (NodeAnnotation x y) = (unpackFS x, unpackFS y)
+
+
+-- | Follow evidence uses under the given node back to their instance bindings,
+-- and connect the declaration to those bindings.
+followEvidenceUses :: ( MonadState Analysis m, MonadReader AnalysisInfo m ) => HieAST a -> Declaration -> m ()
+followEvidenceUses n d = do
+  Config{ typeClassRoots } <- asks weederConfig
+  AnalysisInfo{ refMap } <- ask
+
+  let getEvidenceTrees = mapMaybe (getEvidenceTree refMap)
```

Comment:
This doesn't seem to depend on anything but `AnalysisInfo`, so I wonder if it should go into `AnalysisInfo` itself, rather than be constantly recomputed.

Reply:
Do you mean `getEvidenceTrees`? It's applied to `names` in `evidenceInfos`, which depends on the node given to `followEvidenceUses`.

Reply:
Yea, but the definition of `getEvidenceTrees` only depends on `AnalysisInfo` - right?

Reply:
I think this would be inlined into `evidenceInfos` as such either way:

```haskell
evidenceInfos = concatMap Tree.flatten $ mapMaybe (getEvidenceTree refMap) names
```

so this shouldn't make any difference.

Reply:
Oh, ok, I get it now. `getEvidenceTrees` is itself a function of `Maybe something` so yes, sharing `mapMaybe (getEvidenceTree refMap)` is pointless. I assume `getEvidenceTree` is cheap to call, in which case this comment can be discarded.